src/db.ts

```diff
@@ -70,6 +70,25 @@ export class StudentGradesDB extends Dexie {
       reading_grades: '++id, student_id',
     })
   }
+  async getStudentById(studentId: number): Promise<Student | undefined> {
+    return this.students.get(studentId);
+  }
+
+  async getFictionalGradeByStudentId(studentId: number): Promise<FictionalGrade | undefined> {
+    return this.fictional_grades.where('student_id').equals(studentId).first();
+  }
+
+  async getInformationalGradeByStudentId(studentId: number): Promise<InformationalGrade | undefined> {
+    return this.informational_grades.where('student_id').equals(studentId).first();
+  }
+
+  async getSpellingGradeByStudentId(studentId: number): Promise<SpellingGrade | undefined> {
+    return this.spelling_grades.where('student_id').equals(studentId).first();
+  }
+
+  async getReadingLevelGradeByStudentId(studentId: number): Promise<ReadingLevelGrade | undefined> {
+    return this.reading_grades.where('student_id').equals(studentId).first();
+  }
```

Comment:
I generally like to reserve creating new functions for actions that extend. the functionality of preexisting methods. It seems to me that this code

```js
db.fictional_grades.where('student_id').equals(studentId).first()
```
is just as clear as

```js
getFictionalGradesByStudentId(studentId)
```
and only slightly less cumbersome to write. But if we do create two ways to write the same thing, it might lead to inconsistency when we are trying to refactor.

On top of that, I was thinking of changing up the schema such that each grade has a joint primary key of "[session_id+student_id]". You would create a session for each new test you are trying to grade and I think this creates a good intuitive dependency. In that case, we could query a grade like this
```js
db.fictional_grades.get([session_id, student_id])
```